### PR TITLE
Add fileinput .pyi for Python 2.7.

### PR DESCRIPTION
**Pull request by olliewild**
This change adds type declarations for the 2.7 version of fileinput.

### CODE REVIEW

File: stdlib/2.7/fileinput.pyi [hunk 1]
```diff
@@ -0,0 +1,46 @@
+from typing import Iterable, Callable, IO, Union, Iterator
+
+class FileInput(Iterable[str]):
+    def __init__(
+        self,
+        files: Union[str, Iterable[str]]=None,
```
Comment (gvanrossum) [inline stdlib/2.7/fileinput.pyi]: Can you add an explicit `Optional[]` to the type here?

File: stdlib/2.7/fileinput.pyi [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    def readline(self) -> str: ...
+    def filename(self) -> Union[str, None]: ...
+    def lineno(self) -> int: ...
+    def filelineno(self) -> int: ...
+    def fileno(self) -> int: ...
+    def isfirstline(self) -> bool: ...
+    def isstdin(self) -> bool: ...
+
+def input(
+    files: Union[str, Iterable[str]]=None,
```
Comment (gvanrossum) [inline stdlib/2.7/fileinput.pyi]: This needs `Optional[]` too.

File: stdlib/2.7/fileinput.pyi [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        backup: str=...,
+        bufsize: int=...,
+        mode: str=...,
+        openhook: Callable[[str, str], IO[str]]=...
+        ) -> None: ...
+
+    def __del__(self) -> None: ...
+    def close(self) -> None: ...
+    def __iter__(self) -> Iterator[str]: ...
+    def __getitem__(self, i) -> str: ...
```
Comment (gvanrossum) [inline stdlib/2.7/fileinput.pyi]: No type for `i`?

**Comment by olliewild** [comment]
Thanks.  I believe I've addressed all your comments.

Please take another look.

**Comment by gvanrossum** [comment]
Yup! Sorry for the stern review. :-)

**Pull request merged**